storage/remote/otlptranslator/prometheus/normalize_name.go

```diff
@@ -175,6 +183,93 @@ func normalizeName(metric pmetric.Metric, namespace string) string {
 	return normalizedName
 }
 
+// fieldsFunc is a copy of strings.FieldsFunc from the Go standard library,
+// but it also returns the separators as part of the result.
+func fieldsFunc(s string, f func(rune) bool) ([]string, []string) {
```

Comment:
were there any alternatives you considered to copying code from the go standard library?  We may also need to put these functions in a separate file, and add the go copyright header (in addition to the prometheus one).

Reply:
Interesting, I didn't know it worked like that. It's not an exact copy since I've made small changes to the function, is it needed anyway?

Reply:
Yes.

Reply:
I'll update this during the weekend :)